crypto/rsa/rsa_sign.c

```diff
@@ -96,11 +218,11 @@ int RSA_sign(int type, const unsigned char *m, unsigned int m_len,
         encoded = tmps;
     }
 
-    if (encoded_len > RSA_size(rsa) - RSA_PKCS1_PADDING_SIZE) {
+    if (encoded_len > (size_t)RSA_size(rsa) - RSA_PKCS1_PADDING_SIZE) {
```

Comment:
Should be `encoded_len + RSA_PKCS1_PADDING_SIZE > (size_t) RSA_size(rsa)`

Reply:
Have rearranged - but it is the same thing :) - just a different way of expressing it.

Reply:
Maybe I am confused but if RSA_size() returns number smaller than RSA_PKCS1_PADDING_SIZE then the subtraction (with unsigned size_t) will give a big value and the condition will be false although it should be true by intention.

Reply:
@t8m I do not think you are not confused

Reply:
Fair enough..

statements with multiple nots in them are also confusing :)

Reply:
Hehe, I am now really confused by @kaduk 's  statement. :confused: :laughing:

Reply:
Oops, how confusing.  I meant to agree with you about the unsigned overflow :laughing: